### PR TITLE
[WIP] fix for multibyte language

### PR DESCRIPTION
**Pull request by knorth55**

### CODE REVIEW

File: smach_viewer/src/smach_viewer/xdot/wxxdot.py [hunk 1]
```diff
@@ -523,7 +523,10 @@ def set_dotcode(self, dotcode, filename='<stdin>'):
   def set_xdotcode(self, xdotcode):
     """Set xdot code."""
     #print xdotcode
-    parser = MyXDotParser(bytes(str(xdotcode).encode("utf-8")))
+    if sys.version_info[0] >= 3:
```
Comment (k-okada) [inline smach_viewer/src/smach_viewer/xdot/wxxdot.py:526]: do we need this?
>     parser = MyXDotParser(bytes(str(xdotcode).encode("utf-8")))
works with melodic, if we do not include #40

Comment (knorth55) [reply smach_viewer/src/smach_viewer/xdot/wxxdot.py:526]: without https://github.com/ros-visualization/executive_smach_visualization/pull/40, we cannot show long multibyte language.
I think it is better to support https://github.com/ros-visualization/executive_smach_visualization/pull/40 .

```
Exception in thread Thread-5:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/home/knorth55/ros/melodic/src/ros-visualization/executive_smach_visualization/smach_viewer/scripts/smach_viewer.py", line 953, in _update_graph
    self.set_dotcode(dotstr,zoom=False)
  File "/home/knorth55/ros/melodic/src/ros-visualization/executive_smach_visualization/smach_viewer/scripts/smach_viewer.py", line 972, in set_dotcode
    if self.widget.set_dotcode(dotcode, None):
  File "/home/knorth55/ros/melodic/src/ros-visualization/executive_smach_visualization/smach_viewer/src/smach_viewer/xdot/wxxdot.py", line 499, in set_dotcode
    self.set_xdotcode(xdotcode)
  File "/home/knorth55/ros/melodic/src/ros-visualization/executive_smach_visualization/smach_viewer/src/smach_viewer/xdot/wxxdot.py", line 530, in set_xdotcode
    self.graph = parser.parse()
  File "/home/knorth55/ros/melodic/src/ros-visualization/executive_smach_visualization/smach_viewer/src/smach_viewer/xdot/xdot.py", line 1304, in parse
    DotParser.parse(self)
  File "/home/knorth55/ros/melodic/src/ros-visualization/executive_smach_visualization/smach_viewer/src/smach_viewer/xdot/xdot.py", line 1096, in parse
    self.parse_graph()
  File "/home/knorth55/ros/melodic/src/ros-visualization/executive_smach_visualization/smach_viewer/src/smach_viewer/xdot/xdot.py", line 1105, in parse_graph
    self.parse_stmt()
  File "/home/knorth55/ros/melodic/src/ros-visualization/executive_smach_visualization/smach_viewer/src/smach_viewer/xdot/xdot.py", line 1138, in parse_stmt
    self.parse_subgraph()
  File "/home/knorth55/ros/melodic/src/ros-visualization/executive_smach_visualization/smach_viewer/src/smach_viewer/xdot/xdot.py", line 1119, in parse_subgraph
    self.parse_stmt()
  File "/home/knorth55/ros/melodic/src/ros-visualization/executive_smach_visualization/smach_viewer/src/smach_viewer/xdot/xdot.py", line 1138, in parse_stmt
    self.parse_subgraph()
  File "/home/knorth55/ros/melodic/src/ros-visualization/executive_smach_visualization/smach_viewer/src/smach_viewer/xdot/xdot.py", line 1119, in parse_subgraph
    self.parse_stmt()
  File "/home/knorth55/ros/melodic/src/ros-visualization/executive_smach_visualization/smach_viewer/src/smach_viewer/xdot/xdot.py", line 1154, in parse_stmt
    self.handle_node(id, attrs)
  File "/home/knorth55/ros/melodic/src/ros-visualization/executive_smach_visualization/smach_viewer/src/smach_viewer/xdot/xdot.py", line 1278, in handle_node
    shapes.extend(parser.parse())
  File "/home/knorth55/ros/melodic/src/ros-visualization/executive_smach_visualization/smach_viewer/src/smach_viewer/xdot/xdot.py", line 705, in parse
    t = s.read_text()
  File "/home/knorth55/ros/melodic/src/ros-visualization/executive_smach_visualization/smach_viewer/src/smach_viewer/xdot/xdot.py", line 609, in read_text
    res = res.decode('utf-8')
  File "/usr/lib/python2.7/encodings/utf_8.py", line 16, in decode
    return codecs.utf_8_decode(input, errors, True)
UnicodeDecodeError: 'utf8' codec can't decode byte 0x81 in position 0: invalid start byte
```

**Comment by k-okada** [comment]
Oh, I mean if we need 
+    if sys.version_info[0] >= 3:

for multibyte data, we should include this within [ros-visualization#40](https://github.com/ros-visualization/executive_smach_visualization/pull/40)

**Comment by knorth55** [comment]
this PR https://github.com/ros-visualization/executive_smach_visualization/pull/40 needed when we set low label width like `40` for long japanese text.
So, when we set high lable width like `100`, we can show japanese text in smach_viewer as the image below, in `origin/melodic-devel` branch.
(But all node names are one long line)

Screenshot 2022-07-22 20:03:28

But with `k-okada/noetic` branch, we cannot show long japanese text with high label width like `100`.
We get the following error, which means that the noetic PR completely breaks japanese support for melodic

```
Exception in thread Thread-6:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/home/knorth55/ros/melodic/src/ros-visualization/executive_smach_visualization/smach_viewer/scripts/smach_viewer.py", line 953, in _update_graph
    self.set_dotcode(dotstr,zoom=False)
  File "/home/knorth55/ros/melodic/src/ros-visualization/executive_smach_visualization/smach_viewer/scripts/smach_viewer.py", line 971, in set_dotcode
    if self.widget.set_dotcode(dotcode, None):
  File "/home/knorth55/ros/melodic/src/ros-visualization/executive_smach_visualization/smach_viewer/src/smach_viewer/xdot/wxxdot.py", line 499, in set_dotcode
    self.set_xdotcode(xdotcode)
  File "/home/knorth55/ros/melodic/src/ros-visualization/executive_smach_visualization/smach_viewer/src/smach_viewer/xdot/wxxdot.py", line 526, in set_xdotcode
    parser = MyXDotParser(bytes(str(xdotcode).encode("utf-8")))
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe4 in position 2278: ordinal not in range(128)
```

So I think we should need the `if sys.version_info[0] >= 3:` to support japanese one long line text in `smach_viewer`

**Comment by knorth55** [comment]
I edited https://github.com/ros-visualization/executive_smach_visualization/pull/40 to add more explanation and examples.

**Pull request merged**